### PR TITLE
fix(release): close v0.2.0 readiness gaps

### PR DESCRIPTION
**Pull request by gnanirahulnutakki**
## Outcome

Closes the verified v0.2.0 release-readiness gaps without changing the release version or bypassing any gate.

- keeps the paired agent-recognition benchmark valid during dev-to-main promotion when main predates the daemon
- replaces yanked build 1.5.1 with reviewed build 1.5.0
- constrains the dev dependency graph to pyasn1 0.6.4 or newer within the compatible minor line
- removes cyclic ardur-verify console-entry wiring
- tightens descriptor and cleanup paths identified by CodeQL
- validates the dated changelog contract and keeps generated source docs deterministic
- replaces the nonexistent testing command with maintained release gates
- adds regressions for every repaired path

## Local verification

- commit hooks: all pass, including private-key and hardcoded-secret detection
- Python full suite: 2,099 passed, 33 skipped, 85% coverage
- final changed-module rerun: 260 passed, 1 skipped
- Go: all 27 packages pass; go vet passes; four AuditBench packages pass under race detection
- recognition corpus: all 36 maintained samples pass
- govulncheck: no reachable or imported vulnerabilities
- package: wheel and sdist validation, Twine checks, Python 3.10 and 3.13 install/lifecycle smokes, Python 3.9 rejection guard
- dependency audit: Python 3.10 and 3.13 dev graphs report zero advisories after the pyasn1 constraint
- docs/static: 123 source pages, 133 mirrored artifacts, Hugo, actionlint, and 615-link Lychee audit pass
- plugin: Claude plugin validation and five-receipt smoke pass; browser JavaScript syntax and auth-header checks pass
- Docker proxy: 11 lifecycle checks pass under non-root/read-only/cap-drop/no-new-privileges controls
- Compose: full SPIRE/proxy/hub stack healthy; 11 lifecycle checks pass; clean teardown verified
- seccomp enforcement: enforce returns EPERM, permissive control returns ECONNREFUSED, both evidence chains verify and attestation digests match
- Trivy release policy: no fixable HIGH/CRITICAL records and zero secret findings; full residual findings are explicitly recorded in #362 and are not described as a clean scan

## Tracked follow-ups

- #361 makes the repository-wide Ruff and ShellCheck quality target green before main promotion. These findings predate this branch and are deliberately kept out of this focused repair.
- #362 tracks the 12 distinct unfixed HIGH/CRITICAL Debian advisories, exact applicability evidence, rescan requirements, and required human residual-risk disposition.

## Promotion context

Supports #140, #147, and the overall audit in #181. Promotion from dev to main and release publication remain human-gated. No main branch, release tag, or protected environment was modified by this PR.

## Summary by CodeRabbit

- **Security**
  - Fail-closed governance handling and stricter empty/whitespace validation across workflows.
  - Hardening for dependency/version constraints and safer cleanup failures (including temporary output removal).
- **New Features**
  - Added a dedicated `ardur verify` console-entry wrapper for consistent CLI dispatch.
- **Bug Fixes**
  - Improved agent-recognition benchmark reference/promotion selection and corrected benchmark snapshot/data flow.
  - Fixed release validation, source mirroring exclusions, packaging/tooling pinning, and local evidence/cleanup edge cases.
- **Documentation**
  - Updated changelog/release-evidence notes and expanded the local runnable validation workflow.
- **Tests**
  - Added/strengthened coverage for release gates, promotion logic, benchmark and runtime cleanup, CLI validation, and workflow alignment.

### CODE REVIEW

File: site/content/source/CHANGELOG.md [hunk 1]
```diff
@@ -29,9 +29,17 @@ All notable changes to Ardur will be documented in this file.
...
-## [0.2.0] — 2026-07-19
+## [0.2.0] — 2026-07-22
 
 ### Security
+- Fail closed on every Claude Code `PreToolUse` processing error instead of
+  allowing an action to continue after governance fails
+- Update `golang.org/x/text` to the reviewed CVE-fixed release
+- Constrain the published Python `dev` extra to `pyasn1>=0.6.4,<0.7`, excluding
+  the network denial-of-service flaws in CVE-2026-59884, CVE-2026-59885, and
+  CVE-2026-59886
```
[the remark this answers is not in the document]
Comment (gnanirahulnutakki) [reply site/content/source/CHANGELOG.md]: Addressed in signed DCO commit 893b32a. `docs/release-evidence-v0.2.0.md` now records primary NVD/PyPI sources, an exact Python 3.10/3.13 pip-audit reproduction, an executable PyPI JSON yank-state check, the 2026-07-22 results, and the explicit limitation that offline tests do not independently attest current advisory/yank metadata. CHANGELOG links the verifier path, a regression requires the evidence URLs/limitation, the live PyPI JSON command passes, 16/16 package tests pass, and generated source docs are synchronized.

File: python/tests/test_python_package_release.py [hunk 2]
```diff
@@ -40,6 +48,108 @@ def _project_config() -> dict[str, object]:
...
+    locked_versions = {
+        package["name"]: package["version"] for package in lock["package"]
+    }
+
+    assert PYASN1_SECURITY_FLOOR in config["project"]["optional-dependencies"]["dev"]
+    assert tuple(int(part) for part in locked_versions["pyasn1"].split(".")) >= (
+        0,
+        6,
+        4,
+    )
```
[the remark this answers is not in the document]
Comment (gnanirahulnutakki) [reply python/tests/test_python_package_release.py]: Addressed in signed DCO commit 893b32a. The regression now parses the locked version once and enforces the complete declared interval `(0, 6, 4) <= locked_version < (0, 7)`. Focused package tests pass: 16/16; all repository commit hooks pass.

File: docs/release-evidence-v0.2.0.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        payload = json.load(response)
+    states = [bool(item["yanked"]) for item in payload["urls"]]
+    if not states:
+        raise SystemExit(f"build {version} has no distribution files")
+    return states
+
+
+assert not any(yank_states("1.5.0"))
+assert all(yank_states("1.5.1"))
+print("build 1.5.0 non-yanked; build 1.5.1 yanked")
```
[the remark this answers is not in the document]
Comment (gnanirahulnutakki) [reply docs/release-evidence-v0.2.0.md]: Addressed in signed DCO commit bea4a0a. The published verifier now uses explicit `SystemExit` failure paths instead of optimization-removable assertions, and the generated source mirror was regenerated. The exact embedded code block succeeds under both `python3 -` and `python3 -O -`; 16/16 focused package tests, source-sync verification, and all commit hooks pass.

**Comment by gnanirahulnutakki** [comment]
Final pre-merge qualification for head bea4a0a104cde3db6d3c181f61a456ed65bbb2ae:

- GitHub: 42 successful checks, 6 expected release-only skips, 0 failures, 0 pending.
- CodeRabbit: successful; all 3 findings addressed and all review threads resolved.
- Local Python: full suite 2,099 passed / 33 skipped; final changed-module rerun 260 passed / 1 skipped; package-release checks 16 passed.
- Local Go: all 27 packages passed; vet, selected race suites, recognition corpus (36/36), and Windows cross-build passed.
- Packaging: wheel/sdist, Twine, Python 3.10/3.13 install/lifecycle, and Python 3.9 rejection passed.
- Supply chain: pinned dependency audits reported zero advisories; govulncheck reported zero reachable/imported vulnerabilities; history/current secret scans passed.
- OCI: hardened local runtime and Compose smokes passed. Trivy findings without available fixes are explicitly parked in #362; the release-policy ignore-unfixed gate passed and is not represented as a clean scan.
- Docs/formats: Hugo, link check, source mirror synchronization, JSON/YAML/schema validation, action lint, and shell syntax passed.

The exact head is DCO-signed across all three source commits. Squash merge will include a Signed-off-by trailer. Cluster qualification is a later pre-main gate and is not being claimed here.
